app/assets/javascripts/components/components/status_content.jsx

```diff
@@ -93,7 +93,7 @@ const StatusContent = React.createClass({
     const { status } = this.props;
     const { hidden } = this.state;
 
-    const content = { __html: emojify(status.get('content')) };
+    const content = { __html: emojify(status.get('content').replace(/\s*(<p>|<\/p>)\s*/g, '$1').replace(/\s{2}/g, ' &nbsp;')) };
```

Comment:
Maintains backwards compatibility with earlier versions of mastodon since not everyone is going to upgrade at the same time. Using an HTML parser seemed like overkill since we're just stripping whitespace next to certain patterns, not traversing a DOM or anything.

Using `&nbsp;` allows users to continue using custom spacing without relying on css `pre-wrap` which treats other characters besides ` ` as semantic.